components/brave_wallet_ui/components/desktop/card-headers/panel-action-header.tsx

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { useHistory } from 'react-router-dom'
+
+// Types
+import { WalletRoutes } from '../../../constants/types'
+
+// Utils
+import { openWalletRouteTab } from '../../../utils/routes-utils'
+
+// Styled Components
+import {
+  Button,
+  ButtonIcon,
+  LeftRightContainer
+} from './shared-panel-headers.style'
+import { Row } from '../../shared/style'
+import { HeaderTitle } from './shared-card-headers.style'
+
+interface Props {
+  title: string
+  expandRoute: WalletRoutes
+}
+
+export const PanelActionHeader = (props: Props) => {
+  const { title, expandRoute } = props
+
+  // Routing
+  const history = useHistory()
+
+  // Methods
+  const onClose = () => {
+    if (history.length > 1) {
+      history.goBack()
+      return
+    }
+    history.push(WalletRoutes.PortfolioAssets)
+  }
```

Comment:
I think you need to check location.key instead
https://stackoverflow.com/a/70532858